velox/type/Filter.cpp

```diff
@@ -458,6 +470,75 @@ bool MultiRange::testBytesRange(
   return false;
 }
 
+std::unique_ptr<Filter> MultiRange::mergeWith(const Filter* other) const {
+  switch (other->kind()) {
+    case FilterKind::kMultiRange: {
+      bool bothNullAllowed = nullAllowed_ && other->testNull();
+      // TODO: verify nanAllowed with the other filter.
+      const MultiRange* multiRangeOther = static_cast<const MultiRange*>(other);
+      std::vector<std::unique_ptr<Filter>> merged;
+      for (auto const& filter : this->filters()) {
+        for (auto const& filterOther : multiRangeOther->filters()) {
+          auto innerMerged = filter->mergeWith(filterOther.get());
+          switch (innerMerged->kind()) {
+            case FilterKind::kAlwaysFalse:
+              break;
```

Comment:
Shouldn't this be "continue;"? E.g. (a or b) AND (c or d) = (a and c) or (a and d) or (b and c) or (b and d). If one of these is false, we just drop it but keep the rest.

Reply:
This break is to break the switch, not the nested loop? Thus this is doing what you are saying `If one of these is false, we just drop it but keep the rest.`.